application/src/main/java/dev/ikm/komet/app/App.java

```diff
@@ -901,6 +921,78 @@ private MenuItem createClassicKometMenuItem() {
         return classicKometMenuItem;
     }
 
+    /**
+     * Create a Resource Usage overlay to display metrics
+     *
+     * @return The menu item for launching the resource usage overlay.
+     */
+    private MenuItem createResourceUsageItem() {
+        MenuItem resourceUsageItem = new MenuItem("Resource Usage");
+        KeyCombination classicKometKeyCombo = new KeyCodeCombination(KeyCode.R, KeyCombination.CONTROL_DOWN);
+        resourceUsageItem.setOnAction(actionEvent -> {
+            if (overlayStage != null && overlayStage.isShowing()) {
+                overlayStage.hide();
+            } else {
+                showResourceUsageOverlay();
+            }
+        });
+        resourceUsageItem.setAccelerator(classicKometKeyCombo);
+        return resourceUsageItem;
+    }
+
+    /**
+     * Show the resource usage overlay.
+     */
+    private void showResourceUsageOverlay() {
+        overlayStage = new Stage();
+        overlayStage.initOwner(getFocusedWindow());
+        overlayStage.initModality(Modality.APPLICATION_MODAL);
+        overlayStage.initStyle(StageStyle.TRANSPARENT);
+
+        VBox overlayContent = new VBox();
+        overlayContent.setAlignment(Pos.CENTER);
+        overlayContent.setStyle("-fx-background-color: rgba(0, 0, 0, 0.5); -fx-padding: 20;");
+
+        Label cpuUsageLabel = new Label();
+        Label memoryUsageLabel = new Label();
+        cpuUsageLabel.setTextFill(Color.WHITE);
+        memoryUsageLabel.setTextFill(Color.WHITE);
+        overlayContent.getChildren().addAll(cpuUsageLabel, memoryUsageLabel);
+
+        Scene overlayScene = new Scene(overlayContent, 300, 200, Color.TRANSPARENT);
+        overlayStage.setScene(overlayScene);
+        overlayStage.show();
+
+        // Create and start the timeline to update resource usage
+        resourceUsageTimeline = new Timeline(new KeyFrame(Duration.seconds(1), event -> {
+            updateResourceUsage(cpuUsageLabel, memoryUsageLabel);
+        }));
+        resourceUsageTimeline.setCycleCount(Timeline.INDEFINITE);
+        resourceUsageTimeline.play();
+    }
+
+    /**
+     * Update the resource usage labels with CPU and memory usage.
+     *
+     * @param cpuUsageLabel the label to be used for the CPU usage
+     * @param memoryUsageLabel the label to be used for the memory usage
+     */
+    private void updateResourceUsage(final Label cpuUsageLabel, final Label memoryUsageLabel) {
+        java.lang.management.OperatingSystemMXBean osBean = ManagementFactory.getOperatingSystemMXBean();
+        double cpuLoad = -1;
+        if (osBean instanceof OperatingSystemMXBean) {
+            cpuLoad = ((OperatingSystemMXBean) osBean).getCpuLoad() * 100;
```

Comment:
Can we use Java's new pattern matching?
```java
if (osBean instanceof OperatingSystemMXBean osMXBean) {
   cpu = osMXBean.getCpuLoad() * 100;
   ...
```

Reply:
Actually I don't think there is a need to check if it is an instance of `java.lang.management.OperatingSystemMXBean` since the object is already of that type.

Reply:
It's confusing, but I actually do.  The one class is from the sun package and the other is from the java package, even though they have the same name.  The sun one is a subclass of the java one and has additional functionality to get the CPU, which isn't available on all systems, so it can't be used on IoT and things like that.  Thus, the java one won't let me get the CPU load, unfortunately, so I have to down convert to get it, and it's always worth a check to see if it is the correct class when down casting.

I hadn't used the enhanced instanceof though, so I incorporated that.  That's pretty nice.